### PR TITLE
NEXT-12733 - Add guide on using custom field in the storefront

### PR DESCRIPTION
**Pull request by leichteckig**
What should be done?
Create a new article on our GitBook instance which explains how to use custom fields in a storefront template.
This can be migrated from our previous documentation.

This article should mention:

* The prerequisite, a working plugin (Refer to the plugin base guide)
* Every other prerequisite you figure out during writing the guide (e.g. a subscriber, knowing how to create a service, a controller, etc.)
* A short code example, including an explanation, on how to do it
* Also explain "Using custom field labels", see in the links below for more information
Category: Extensions > Plugins > Storefront

Are there already guides in the previous documentation for this?
Yes, at least the custom field labels part! But please, don't just copy & paste. Make sure it still works and rewrite it to fit our new writing guidelines!
https://docs.shopware.com/en/shopware-platform-dev-en/how-to/use-custom-field-labels-in-storefront

For more information, ask me, Patrick Stahl.

### CODE REVIEW

File: guides/plugins/plugins/storefront/using-custom-fields-storefront.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
+# Using custom fields in the storefront
+
+## Overview
+
+This guide will show you how to use custom fields, e.g. labels in the storefront.
+
+## Prerequisites
+
+You won't learn to create a plugin in this guide, head over to our plugin base guide to create a plugin first,
```
Comment (raknison) [inline guides/plugins/plugins/storefront/using-custom-fields-storefront.md:9]: maybe add a link to the plugin base guide?

File: guides/plugins/plugins/storefront/using-custom-fields-storefront.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+This guide will show you how to use custom fields, e.g. labels in the storefront.
+
+## Prerequisites
+
+You won't learn to create a plugin in this guide, head over to our plugin base guide to create a plugin first, 
+if you don't know how it's done yet. 
+
+Needless to say, you need a custom field itself to add to the Storefront via your own plugin. Head over to the guide
+on [adding custom fields to Shopware]() to be able to prepare your own custom field.
```
Comment (raknison) [inline guides/plugins/plugins/storefront/using-custom-fields-storefront.md:13]: link missing in `[adding custom fields to Shopware]()`

File: guides/plugins/plugins/storefront/using-custom-fields-storefront.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+{% hint style="info" %}Did you notice the Twig function `sw_sanitize`? It's a Twig function we wrote, customized for Shopware's needs.
+It filters tags and attributes from a given string optimized for Shopware usage.
+{% endhint %}
+
+Imagine you want to add a text field to the product description. If you want to use the snippet in the storefront,
+you have to extend a template file first. Let's say we want to add our custom field to the product description's text. 
+The block of this element is `page_product_detail_description_content_text`, so we'll use it in our example. 
+As we want to add our custom field in there, we use `parent` Twig function to keep the original template:
+
+{% code title="<plugin root>/src/Core/Content/Example/ExampleDefinition.php" %}
```
Comment (raknison) [inline guides/plugins/plugins/storefront/using-custom-fields-storefront.md:41]: wrong title - should be a twig file

File: guides/plugins/plugins/storefront/using-custom-fields-storefront.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+{% block page_product_detail_description_content_text %}
+    {{ parent() }}
+{% endblock %}
+```
+{% endcode %}
+
+Now, we finally add our custom field as explained before:
+
+{% code title="<plugin root>/src/Core/Content/Example/ExampleDefinition.php" %}
```
Comment (raknison) [inline guides/plugins/plugins/storefront/using-custom-fields-storefront.md:53]: some here wrong title

**Review by raknison — APPROVED** [review]

**Review by tobiasberge — APPROVED** [review]

**Pull request merged**